crates/polars-io/src/json/mod.rs

```diff
@@ -221,6 +221,17 @@ where
     schema_overwrite: Option<&'a Schema>,
     json_format: JsonFormat,
 }
+fn remove_bom(bytes: Vec<u8>) -> Vec<u8> {
+    if bytes.starts_with(&[0xEF, 0xBB, 0xBF]) {
+        // UTF-8 BOM
+        bytes[3..].to_vec()
```

Comment:
This wourl reallocate. We should do the before we allocate a vec from the input.